cleanup/artifactCleanup/artifactCleanup.groovy

```diff
@@ -137,3 +156,40 @@ private def artifactCleanup(int months, String[] repos, log, paceTimeMS, dryRun
         log.info "Finished cleanup, deleted $foundArtifacts artifacts that took up $bytesFound bytes"
     }
 }
+
+private def getSkippedPaths(String[] repos) {
+    def timeStart = new Date()
+    def skip = [:]
+    for (String repoKey : repos){
+        def pathsTmp = []
+        def aql = "items.find({\"repo\":\"" + repoKey + "\",\"type\": \"any\",\"@cleanup.skip\":\"true\"}).include(\"path\", \"name\")"
+        searches.aql(aql.toString()) {
+            for (item in it) {
+                def path = item.path + '/' + item.name
+                // Root path case behavior
+                if ("." == item.path){
+                    path = item.name
+                }
+                if (log.isTraceEnabled()){
+                    log.trace "skip found for repo " + repoKey + ": " + path
+                }
+                pathsTmp.add(path)
+            }
+        }
+        // Simplify list to have only parent paths (TODO: perhaps below lines provide poor performance if several thousand of skip)
```

Comment:
@DarthFennec : Nice improvement (I was in a brainless mode pre-weekend ^^).
PR will be updated wednesday (I have not Artifactory pro @home